src/query.rs

```diff
@@ -184,4 +208,612 @@ pub fn scanlog(text: &String, flags: ScanFlags) -> HashMap<String, Vec<String>>
+                                    //TODO: Handle error. Ideally we want to stop the channel
+                                    "".to_string()
+                                }
+                            };
+
+
+                            let individual_lines: Vec<String> = all_file_lines.lines().map(|x| x.to_string()).collect();
+                            let queries_parse = queries_parse.clone();
+                            let query = query.clone();
+                            stream::iter_ok(individual_lines).fold(tx, move |tx, line| {
+                                let mut projection_values: HashMap<String, String> = HashMap::new();
+                                let query_data = queries_parse.get(&query.to_string()[..]).unwrap();
+                                // if we have position columns, process
+                                if query_data.positional_fields.len() > 0 {
+                                    //TODO: Use separator construct from header
+                                    let parts: Vec<&str> = line.split(" ").collect();
+                                    for pos in &query_data.positional_fields {
+                                        if pos.position - 1 < (parts.len() as i32) {
+                                            projection_values.insert(pos.alias.clone(), parts[(pos.position - 1) as usize].to_string());
+                                        } else {
+                                            projection_values.insert(pos.alias.clone(), "".to_string());
+                                        }
+                                    }
+                                }
+
+                                if query_data.smart_fields.len() > 0 {
+                                    let found_vals = scanlog(&line.to_string(), query_data.scan_flags);
+                                    for smt in &query_data.smart_fields {
+                                        if found_vals.contains_key(&smt.typed[..]) {
+                                            // if the requested position is available
+                                            if smt.position - 1 < (found_vals[&smt.typed].len() as i32) {
+                                                projection_values.insert(smt.alias.clone(), found_vals[&smt.typed][(smt.position - 1) as usize].clone());
+                                            } else {
+                                                projection_values.insert(smt.alias.clone(), "".to_string());
+                                            }
+                                        }
+                                    }
+                                }
+
+                                // filter the line
+                                let mut skip_line = false;
+                                match query {
+                                    sqlparser::sqlast::SQLStatement::SQLSelect(ref q) => {
```

Comment:
This code should be simplified, it seems too long and should be converted to much smaller functions.

Reply:
definetively, that's what I'm working on next, since the basic ingest and query are ready, now I'm refactoring the code, starting with `main.rs` to `lib.rs` I'll start breaking the mega functions into smaller chunks in subsequent PRs